### PR TITLE
WIP: Cuda build

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -109,6 +109,7 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
     export GCC_HOST_COMPILER_PREFIX="$(dirname ${GCC})"
     # No obvious way to specify a different ld, so just link it to conda-forge's
     ln -s ${LD} $(dirname ${LD})/ld
+    ln -s ${AR} $(dirname ${LD})/ar
```
Comment (izahn) [inline recipe/build.sh]: This looks like a promising track! If you want to do all binutils

```
    ln -s ${AR} $(dirname ${AR})/ar
    ln -s ${AS} $(dirname ${AS})/as
    ln -s ${CXXFILT} $(dirname ${CXXFILT})/c++filt
    ln -s ${ELFEDIT} $(dirname ${ELFEDIT})/elfedit
    ln -s ${GPROF} $(dirname ${GPROF})/gprof
    ln -s ${LD_GOLD} $(dirname ${LD_GOLD})/ld.gold
    ln -s ${LD} $(dirname ${LD})/ld
    ln -s ${NM} $(dirname ${NM})/nm
    ln -s ${OBJCOPY} $(dirname ${OBJCOPY})/objcopy
    ln -s ${OBJDUMP} $(dirname ${OBJDUMP})/objdump
    ln -s ${RANLIB} $(dirname ${RANLIB})/ranlib
    ln -s ${READELF} $(dirname ${READELF})/readelf
    ln -s ${SIZE} $(dirname ${SIZE})/size
    ln -s ${STRINGS} $(dirname ${STRINGS})/strings
    ln -s ${STRIP} $(dirname ${STRIP})/strip
```
should be pretty comprehensive

**Comment by izahn** [comment]
Just for reverence https://github.com/bazelbuild/bazel/issues/8715 and https://github.com/tensorflow/tensorflow/issues/31083 document previous encounters with this problem.

File: recipe/build.sh (hunk 1 above)
Comment (hmaarrfk) [reply recipe/build.sh]: i could, but....... it seems i'm still hitting the -lsnappy and `-lprotobuf` errors even though conda-forge `ld` was used. Will add your list though.

Comment (njzjz) [reply recipe/build.sh]: You do not need to do so by yourself. You can add `binutils` package, which has added symbolic links here: https://github.com/conda-forge/binutils-feedstock/blob/11da643035ec29d219d7d32a7c8bb6d6ef911d45/recipe/install_binutils_symlinks.sh#L8

**Comment by hmaarrfk** [comment]
Why do we have this file? Is it being used?

**Comment by hmaarrfk** [comment]
I feel like we should be remove the `template` extension

File: recipe/build.sh (hunk 1 above)
Comment (hmaarrfk) [reply recipe/build.sh]: thanks!

**Comment by isuruf** [comment]
Try the following patch,
```diff
--- ./third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl      2021-06-22 16:25:00.722235223 -0500
+++ ./third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl      2021-06-22 16:27:20.303397144 -0500
@@ -245,6 +245,8 @@
            ' --compiler-bindir=' + GCC_HOST_COMPILER_PATH +
            ' -I .' +
            ' -x cu ' + opt + includes + ' ' + srcs + ' -M -o ' + depfile)
+    cmd += ' -L' + PREFIX_DIR + '/../lib'
+    cmd += ' -Wl,-rpath,' + PREFIX_DIR + '/../lib'
     if log: Log(cmd)
     exit_status = system(cmd)
     if exit_status != 0:
@@ -255,6 +257,8 @@
          ' --compiler-bindir=' + GCC_HOST_COMPILER_PATH +
          ' -I .' +
          ' -x cu ' + opt + includes + ' -c ' + srcs + out)
+  cmd += ' -L' + PREFIX_DIR + '/../lib'
+  cmd += ' -Wl,-rpath,' + PREFIX_DIR + '/../lib'
 
   # TODO(zhengxq): for some reason, 'gcc' needs this help to find 'as'.
   # Need to investigate and fix.
@@ -282,7 +286,8 @@
   # this).
   cpu_compiler_flags = [flag for flag in sys.argv[1:]
                              if not flag.startswith(('--cuda_log'))]
-
+  cpu_compiler_flags += ["-L", PREFIX_DIR + "/../lib"]
+  cpu_compiler_flags += ["-Wl,-rpath", PREFIX_DIR + "/../lib"]
   return subprocess.call([CPU_COMPILER] + cpu_compiler_flags)
 
 if __name__ == '__main__':
 ```

**Pull request closed without merging**